### PR TITLE
implemented searchfield for language book selection

### PR DESCRIPTION
**Pull request by kievu**
Fixes https://github.com/GlobalDigitalLibraryio/issues/issues/501

It looks like this: https://puu.sh/CbBtj/6a0d3959d3.png

### CODE REVIEW

File: packages/gdl-frontend/components/LanguageList.js [hunk 1]
```diff
@@ -30,56 +32,121 @@ type Props = {
...
+        )}
+      </List>
+    );
+  }
+}
+
+const styles = {
+  visibleScrollbar: css`
+    overflow-y: scroll;
+    width: 300px;
```
Comment (kievu) [inline packages/gdl-frontend/components/LanguageList.js]: When searching, the list changes and renders it widths according to the longest string. I sat the width to 300px to make it consistent.

File: packages/gdl-frontend/components/LanguageList.js [hunk 2]
```diff
@@ -30,56 +32,121 @@ type Props = {
...
+    };
+  }
+
+  searchLanguage = (
+    value: SyntheticInputEvent<EventTarget>,
+    filteredLanguages: Array<Language>
+  ) => {
+    const input = value.target.value.toLowerCase();
+
+    const result = filteredLanguages.filter(element =>
```
Comment (vikre) [inline packages/gdl-frontend/components/LanguageList.js]: Would have renamed `element` to `language`

File: packages/gdl-frontend/components/LanguageList.js [hunk 3]
```diff
@@ -30,56 +32,121 @@ type Props = {
...
+    return selectedLanguage
+      ? languages.filter(l => l !== selectedLanguage)
+      : languages;
+  };
+
+  render() {
+    const { onSelectLanguage, linkProps } = this.props;
+    const { searchResult } = this.state;
+
+    const selectedLanguage = this.getSelectedLangauge();
```
[the remark this answers is not in the document]
Comment (kievu) [reply packages/gdl-frontend/components/LanguageList.js]: Skrevet feil mange ganger på `langauge` :D

**Comment by kievu** [comment]
searchmobile
searchdesktop

File: packages/gdl-frontend/components/LanguageList.js (hunk 1 above)
Comment (kievu) [reply packages/gdl-frontend/components/LanguageList.js]: Changed to dynamically set width based on initial width when opened.

File: packages/gdl-frontend/components/LanguageList.js [hunk 4]
```diff
@@ -14,72 +14,157 @@ import {
...
+    const { enableSearch, onSelectLanguage, linkProps } = this.props;
+    const { searchResult } = this.state;
+
+    const selectedLanguage = this.getSelectedLanguage();
+    const filteredLanguages = this.getFilteredLanguages(selectedLanguage);
+    const noResult = searchResult.length === 0;
+
+    return (
+      <List
+        id="listContainer"
```
[the remark this answers is not in the document]
Comment (kievu) [reply packages/gdl-frontend/components/LanguageList.js]: Bra forslag! Fikset ;)

**Pull request merged**